src/utils/permission/index.js

```diff
@@ -93,6 +93,7 @@ export const requestExternalStoragePermission = async () => {
         break;
 
       case RESULTS.GRANTED:
+      case RESULTS.LIMITED:
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `RESULTS.LIMITED` case is a good improvement for handling different permission states. However, it's important to note that the `LIMITED` status is specific to iOS 14 and later. It allows users to select specific photos and videos for an app to access. If your application needs full gallery access, you might want to handle this case differently than `GRANTED`. 
```diff
       case RESULTS.GRANTED:
+      case RESULTS.LIMITED:
         message = StringConstant.cameraPermissionGranted;
         success = true;
         break;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->